[0.6.x] MacOS: Add experimental ARM64 build and release support

AkiSakurai opened the pull request:
This commit introduces comprehensive support for building, testing, and releasing OpenTabletDriver natively for macOS on the ARM64 architecture (Apple Silicon).
  Key changes include:
  CI/CD:
  
  Added a macospublish-arm64 job to dotnet.yml to build and upload ARM64 artifacts.
  Added a macos_arm64 job to release.yml to create and upload ARM64 release packages.
  
  Build System:
  
  The eng/macos/package.sh script now auto-detects the architecture (arm64 or x64) and builds for the corresponding runtime.
  
  The script performs ad-hoc codesigning on the application bundle for both ARM64 and x64 binaries.
  Codesigning is required for ARM64 binaries to run on Apple Silicon but is not required for x64 binaries; however, codesigning is applied to both for consistency.
  
  An earlier version of Eto had a bug where the automatic bundle was not generated when an output folder was specified.
  This bug has been fixed in Eto version 2.10.2, which now generates the bundle automatically.
  Since we manually create the application bundle, this commit disables Eto's automatic .app creation during publishing in the build script to prevent nested bundle creation.
  
  Project Configuration:
  
  Upgraded Eto.Platform.Mac64 to version 2.10.2.
  This version includes critical fixes for UI rendering, stability, universal binary compatibility on ARM64, and resolves the issue with extraneous bundle creation.
  
  Updated OpenTabletDriver.UX.MacOS.csproj to include resources like the app icon and Info.plist for Eto's automatic bundle generation.
  
  Added UseCurrentRuntimeIdentifier to the project configuration to allow Rider to build and execute using the host's architecture.

gonX commented:
Sadly, some of the build and CI stuff has improved massively on the `avalonia` branch with #3791, and due to the scope of these changes, they aren't planned to be included on this branch before *after* the release of 0.6.6.

I suggest that you split the following 2 features into a separate PR, which can then be included within the 0.6.6 merge window:

- `.run` xml files
- launchSettings
- codesign, if you deem it necessary

AkiSakurai committed:
[0.6.x] MacOS: Add native ARM64 build and release support
    This commit introduces comprehensive support for building, testing, and releasing OpenTabletDriver natively for macOS on the ARM64 architecture (Apple Silicon).
    Key changes include:
    CI/CD:

    Added a arm64 job to dotnet.yml to build and upload ARM64 artifacts.

    Build System:

    The script performs ad-hoc codesigning on the application bundle for both ARM64 and x64 binaries.
    Codesigning is required for ARM64 binaries to run on Apple Silicon but is not required for x64 binaries; however, codesigning is applied to both for consistency.

    An earlier version of Eto had a bug where the automatic bundle was not generated when an output folder was specified.
    This bug has been fixed in Eto version 2.10.1, which now generates the bundle automatically.
    Since we manually create the application bundle, this commit disables Eto's automatic .app creation during publishing in the build script to prevent nested bundle creation.

    Project Configuration:

    Upgraded Eto.Platform.Mac64 to version 2.11.0.
    This version includes critical fixes for UI rendering, stability, universal binary compatibility on ARM64, and resolves the issue with extraneous bundle creation.

    Updated OpenTabletDriver.UX.MacOS.csproj to include resources like the app icon and Info.plist for Eto's automatic bundle generation.

    Added UseCurrentRuntimeIdentifier to the project configuration to allow Rider to build and execute using the host's architecture.

gonX commented:
Can be integrated now if you want to rebuild this on top of the new build system. It should already work with `./build.sh --runtime osx-arm64` but `codesign` still needs adding.

As Mrcubix mentioned, Eto has some regressions with some GUI features on Linux and Windows. For example, multi-tablet users being on the filters list does not get correctly updated when selected tablet is disconnected, and it falls over to next tablet in line. You should check if the macOS version is affected by this.

Also see #2336 and #3655

AkiSakurai commented:
> As Mrcubix mentioned, Eto has some regressions with some GUI features on Linux and Windows. For example, multi-tablet users being on the filters list does not get correctly updated when selected tablet is disconnected, and it falls over to next tablet in line. You should check if the macOS version is affected by this.

I don't observe that issue, but the layout of the About page is broken.

> Also see #2336 and #3655

gonX reviewed (dismissed):
LGTM, but I am not merging before you've undrafted the PR.

gonX reviewed (approved):
LGTM

dsarno commented:
Tested this branch on Apple Silicon and can confirm the GUI launches and works natively on ARM64.

**Test environment:**
- MacBook Pro (MacBookPro18,2) — M1 Pro
- macOS 26.3 (arm64)
- XP-Pen Deco 01 V3

**Results:**
- GUI launches without the `objc_msgSend_stret` crash that occurs with Eto.Platform.Mac64 2.5.10
- Tablet detected successfully
- UI renders and interacts correctly
- Pen input and express key bindings work

Previously we were running the daemon natively on ARM64 with CLI-only configuration as a workaround. This PR resolves the GUI gap.

gonX reviewed (changes requested):
Needs rebase but I like what I'm seeing
